source/client-side-operations-timeout/tests/client-side-encryption.yml

```diff
@@ -0,0 +1,79 @@
+
+tests:
+  - description: "timeoutMS applied to listCollections to get collection schema"
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 1 }
+      data:
+        failCommands: ["listCollections"]
+        blockConnection: true
+        blockTimeMS: 15
+    clientOptions:
+      autoEncryptOpts:
+        kmsProviders:
+          aws: {} # Credentials filled in from environment.
+      timeoutMS: 10
+    operations:
+      - name: insertOne
+        arguments:
+          document: &doc0 { _id: 1, encrypted_string: "string0", random: "abc" }
+        result:
+          timeoutError: true
+    expectations:
+      # Auto encryption will request the collection info.
+      - command_started_event:
+          command:
+            listCollections: 1
+            filter:
+              name: *collection_name
+          command_name: listCollections
+  - description: "timeoutMS applied to find to get keyvault data"
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 1 }
+      data:
+        failCommands: ["find"]
+        blockConnection: true
+        blockTimeMS: 15
+    clientOptions:
+      autoEncryptOpts:
+        kmsProviders:
+          aws: {} # Credentials filled in from environment.
+        timeoutMS: 10
+    operations:
+      - name: insertOne
+        arguments:
+          document: *doc0
+        result:
+          timeoutError: true
+    expectations:
+      # Auto encryption will request the collection info.
+      - command_started_event:
+          command:
+            listCollections: 1
+            filter:
+              name: *collection_name
+          command_name: listCollections
+      - command_started_event:
+          command:
+            listCollections: 1
+            filter:
+              name: "datakeys"
+            $db: keyvault
+          command_name: listCollections
+      # Then key is fetched from the key vault.
+      - command_started_event:
+          command:
+            find: datakeys
+            filter: {"$or": [{"_id": {"$in": [ {'$binary': {'base64': 'AAAAAAAAAAAAAAAAAAAAAA==', 'subType': '04'}} ] }}, {"keyAltNames": {"$in": []}}]}
+            $db: keyvault
+            readConcern: { level: "majority" }
+          command_name: find
```

Comment:
Each of the `listCollections`, `find` and `insert` commands are all under the same timeout. Is it worth adding a test that blocks each for 10ms, then sets `timeoutMS` to 25ms, to check that the running sum of the timeout is accounted for?

Reply:
Yeah, this should've been done differently. I removed this test and added a different one that blocks `listCollections` and `find` for 10ms each and runs the `insert` with `timeoutMS=25`. In total, the FLE portion should take 30ms (2 `listCollections` and 1 `find`).